rust/lancedb/src/remote/client.rs

```diff
@@ -245,6 +261,49 @@ impl RestfulLanceDbClient<Sender> {
         if let Some(timeout) = timeout {
             client_builder = client_builder.timeout(timeout);
         }
+
+        // Configure mTLS if TlsConfig is provided
+        if let Some(tls_config) = &client_config.tls_config {
+            // Load client certificate and key for mTLS
+            if let (Some(cert_file), Some(key_file)) = (&tls_config.cert_file, &tls_config.key_file)
+            {
+                let cert = std::fs::read(cert_file).map_err(|err| Error::Other {
```

Comment:
You might want to make these async:


```suggestion
                let cert = tokio::fs::read(cert_file).await.map_err(|err| Error::Other {
```

Reply:
the issue is that the whole `try_new` is sync, if we change it to `tokio` it has to propagate all the way up. Since this is just a 1 time read when creating the client at initialization time, I felt it is okay to just make it sync. @wjones127 what do you think?